modules/dnn/src/dnn.cpp

```diff
@@ -135,6 +135,20 @@ class BackendRegistry
     }
 #endif
 
+    static inline bool deviceSupportsFp16() {
```

Comment:
The `inline` isn't required. The function name is not very clear. This function only checks for FP16 support in CUDA capable devices but the name suggests that it checks for any device.

Reply:
Something like `currentCudaDeviceSupportsFp16()`?